source/fe/fe_q_hierarchical.cc

```diff
@@ -124,6 +147,59 @@ FE_Q_Hierarchical<dim>::clone() const
   return new FE_Q_Hierarchical<dim>(*this);
 }
 
+template <int dim>
+void
+FE_Q_Hierarchical<dim>::get_interpolation_matrix(const FiniteElement< dim> &source,
+                                                 FullMatrix< double > &matrix) const
+{
+  // support interpolation between FE_Q_Hierarchical only.
+  if (const FE_Q_Hierarchical<dim> *source_fe
+      = dynamic_cast<const FE_Q_Hierarchical<dim>*>(&source))
+    {
+      // ok, source is a Q_Hierarchical element, so we will be able to do the work
+      Assert (matrix.m() == this->dofs_per_cell,
+              ExcDimensionMismatch (matrix.m(),
+                                    this->dofs_per_cell));
+      Assert (matrix.n() == source.dofs_per_cell,
+              ExcDimensionMismatch (matrix.m(),
+                                    source_fe->dofs_per_cell));
+
+      // make sure that we interpolate to a reacher element.
+      Assert (this->dofs_per_cell >= source_fe->dofs_per_cell,
+              ExcMessage("Interpolation to higher polynomial degree is advised."));
```

Comment:
Why do we need to require this? Can't we just truncate higher modes otherwise?

If you'd like to keep it that way, may I suggest to change to use `ExcNotImplemented` instead, or an exception text that makes it clear that it isn't just a preference ("advised") but a hard requirement that this function can not be used with a coarser element?